src/widgets/histogram/histogram-widget-model.js

```diff
@@ -52,6 +52,19 @@ module.exports = WidgetModel.extend({
     this.set('autoStyle', false);
   },
 
+  setState: function (state) {
+    if (state && state.start && state.end) {
```

Comment:
From the PR desc:
> start and end were needed in order to set the proper bounds.
> …
> start and end should not be requested after the map instantiation, if not filter could change the returned data (@rochoa).
> …
> After a second refresh with histrogram state changes, the histogram values disappear.

Not familiar enough with the histogram model and what it does, I think @javierarce knows that part better. 
That said, from what I can see in the implementation I think part of the problem is related to my [other comment](https://github.com/CartoDB/deep-insights.js/pull/440/files#r74257959), assuming that part would be fixed, the histogram model should setup the initial state (request full start/end range etc.) _regardless_ of if the initial attrs contains a custom range or not. If the attrs contains a custom range it should of course apply that as well.